### PR TITLE
chore: upgrade to polkadot v1.14.0

### PR DESCRIPTION
**Pull request by chungquantin**
Status: ✅ Compiled and built successfully

Prerequisite: https://github.com/r0gue-io/assets-parachain/pull/7

- ISSUE: https://github.com/r0gue-io/base-parachain/issues/39
- Release: https://github.com/paritytech/polkadot-sdk/releases/tag/polkadot-v1.14.0
- Reference `base-parachain`: https://github.com/r0gue-io/base-parachain/pull/41

- Compare [Parachain Template v1.13.0](https://github.com/paritytech/polkadot-sdk/commits/polkadot-v1.13.0/templates/parachain) vs [Parachain Template v1.14.0](https://github.com/paritytech/polkadot-sdk/commits/polkadot-v1.14.0/templates/parachain)

## References
- Compare `pallet-nfts`: [`v1.13.0`](https://github.com/paritytech/polkadot-sdk/commits/polkadot-v1.13.0/substrate/frame/nfts) and [`v1.14.0`](https://github.com/paritytech/polkadot-sdk/commits/polkadot-v1.14.0/substrate/frame/nfts)
- Compare `pallet-assets`: [`v1.13.0`](https://github.com/paritytech/polkadot-sdk/commits/polkadot-v1.13.0/substrate/frame/assets) and [`v1.14.0`](https://github.com/paritytech/polkadot-sdk/commits/polkadot-v1.14.0/substrate/frame/assets)
- Compare `pallet-nft-fractionalization`: [`v1.13.0`](https://github.com/paritytech/polkadot-sdk/commits/polkadot-v1.13.0/substrate/frame/nft-fractionalization) and [`v1.14.0`](https://github.com/paritytech/polkadot-sdk/commits/polkadot-v1.14.0/substrate/frame/nft-fractionalization)

## Direct changes
- Introduce the construct runtime V2 (https://github.com/r0gue-io/base-parachain/issues/37)
- [[#4730](https://github.com/paritytech/polkadot-sdk/pull/4730)]: rpc upgrade jsonrpsee to v0.23.1
- [[#4769](https://github.com/paritytech/polkadot-sdk/pull/4769)]: Use real rust type for pallet alias in `runtime` macro
- Use aggregated types for `RuntimeFreezeReason`: https://github.com/paritytech/polkadot-sdk/commit/988103d7578ad515b13c69578da1237b28fa9f36

### CODE REVIEW

**Review by Daanvdplas — APPROVED** [review]
LGTM!

**Review by al3mart — APPROVED** [review]
<3

File: runtime/src/configs/mod.rs [hunk 1]
```diff
@@ -242,6 +244,10 @@ impl cumulus_pallet_xcmp_queue::Config for Runtime {
     type ControllerOriginConverter = XcmOriginToTransactDispatchOrigin;
     type WeightInfo = (); // Configure based on benchmarking results.
     type PriceForSiblingDelivery = NoPriceForMessageDelivery<ParaId>;
+    // Limit the number of messages and signals a HRML channel can have at most
+    type MaxActiveOutboundChannels = ConstU32<128>;
+    // Limit the number of HRML channels
+    type MaxPageSize = ConstU32<{ 1 << 16 }>;
```
Comment (al3mart) [inline runtime/src/configs/mod.rs:250]: Let's configure `MaxPageSize` as in https://github.com/r0gue-io/pop-node/pull/198#issuecomment-2297673149

**Review by al3mart — CHANGES_REQUESTED** [review]

File: runtime/src/configs/mod.rs (hunk 1 above)
Comment (chungquantin) [reply runtime/src/configs/mod.rs:250]: @al3mart For the pop-node case, it is understandable but for the templates, I think we should follow the code in Polkadot SDK templates. What do you think?

Comment (al3mart) [reply runtime/src/configs/mod.rs:250]: Just thinking that users launching their local environments with `pop up` and the served network file will get a relay with `hrmpChannelMaxMessageSize: 1,048,576` so maybe providing that higher bound could be leveraged by some users.

Comment (chungquantin) [reply runtime/src/configs/mod.rs:250]: Makes sense @al3mart, updated!

**Review by AlexD10S — APPROVED** [review]

**Review by Daanvdplas — APPROVED** [review]

**Pull request merged**